server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java

```diff
@@ -201,8 +214,18 @@ public ActionRequestValidationException validate() {
             if (policies.length != 0) {
                 validationException = addValidationError("can't use slm policy filter with verbose=false", validationException);
             }
-        } else if (after != null && offset > 0) {
-            validationException = addValidationError("can't use after and offset simultaneously", validationException);
+            if (afterValue != null) {
+                validationException = addValidationError("can't use after_value with verbose=false", validationException);
+            }
+        } else if (offset > 0) {
+            if (after != null) {
+                validationException = addValidationError("can't use after and offset simultaneously", validationException);
+            }
+            if (afterValue != null) {
```

Comment:
Is there a reason we are not allowing this? `offset` could just be evaluated after `after_value`?

Obviously we can do that in a follow-up rather than here.

Reply:
Huh yea actually we have to that for Kibana (otherwise any filtered results can't be paginated through which seems weird). Let's push it to a follow-up though as it will require some more code-heavy tests I think, though the production code change is easy/small :)